boa/src/builtins/value/mod.rs

```diff
@@ -762,25 +766,27 @@ impl ValueData {
         }
     }
 
-    /// Get the type of the value
+    /// Get the type of the value.
     ///
-    /// https://tc39.es/ecma262/#sec-typeof-operator
-    pub fn get_type(&self) -> &'static str {
+    /// This is similar to typeof as described at https://tc39.es/ecma262/#sec-typeof-operator but instead of
+    /// returning a string it returns a Type enum which implements fmt::Display to allow getting the string if
+    /// required using to_string().
+    pub fn get_type(&self) -> Type {
         match *self {
-            Self::Rational(_) | Self::Integer(_) => "number",
-            Self::String(_) => "string",
-            Self::Boolean(_) => "boolean",
-            Self::Symbol(_) => "symbol",
-            Self::Null => "object",
-            Self::Undefined => "undefined",
+            Self::Rational(_) | Self::Integer(_) => Type::Number,
+            Self::String(_) => Type::Str,
+            Self::Boolean(_) => Type::Boolean,
+            Self::Symbol(_) => Type::Symbol,
+            Self::Null => Type::Null,
+            Self::Undefined => Type::Undefined,
             Self::Object(ref o) => {
                 if o.deref().borrow().is_callable() {
-                    "function"
+                    Type::Function
                 } else {
-                    "object"
+                    Type::Object
                 }
             }
-            Self::BigInt(_) => "bigint",
+            Self::BigInt(_) => Type::BigInt,
         }
     }
```

Comment:
Maybe, it would be nicer to put `get_type` in `val_type` module

Reply:
Done, commit 93273ae